packages/react-reconciler/src/ReactFiberBeginWork.new.js

```diff
@@ -662,10 +662,13 @@ function updateOffscreenComponent(
         nextBaseLanes = mergeLanes(prevBaseLanes, renderLanes);
         if (enableCache) {
           // Save the cache pool so we can resume later.
+          const prevCachePool = prevState.cachePool;
+          if (prevCachePool !== null) {
```

Comment:
The logic around when to push and pop is too confusing. I was being too clever when I wrote this code originally.

Let's make it so if there's no spawned cache pool we need to restore, instead of pushing nothing, we push the current value to the stack again. Then in complete/unwind, you can just pop again without any additional checks.